packages/rich-text/src/create.js

```diff
@@ -57,6 +57,13 @@ function toFormat( { type, attributes } ) {
 		return attributes ? { type, attributes } : { type };
 	}
 
+	if (
+		formatType.__experimentalCreatePrepareEditableTree &&
+		! formatType.__experimentalCreateOnChangeEditableValue
```

Comment:
Ah, that's a neat pragmatic way to solve it. Nice.

Reply:
Yeah, we have to keep the annotation formats around until the very, very end, right before it's sent out of RichText. Since you'd expect to be able to type inside and outside, you'd also expect it to have the format boundaries.